vi/cs-230-recurrent-neural-networks.md

```diff
@@ -0,0 +1,677 @@
+
+<br>
+
+
+**52. [A cute teddy bear is reading, teddy bear, soft, Persian poetry, art]**
+
+&#10230; [Một chú gấu bông dễ thương đang đọc sách, gấu bông teddy, soft, thơ Persian, hội hoạ]
+
+<br>
+
+
+**53. [Train network on proxy task, Extract high-level representation, Compute word embeddings]**
+
+&#10230; [Huấn luyện mạng trên proxy task, Bóc tách các thể hiện cấp cao, Tính toán word embeddings]
+
+<br>
+
+
+**54. Skip-gram ― The skip-gram word2vec model is a supervised learning task that learns word embeddings by assessing the likelihood of any given target word t happening with a context word c. By noting θt a parameter associated with t, the probability P(t|c) is given by:**
+
+&#10230; Skip-gram - Mô hình skip-gram word2vec là một task supervised learning, nó học các word embeddings bằng cách đánh giá khả năng của bất kì target word t cho trước nào xảy ra với context word c. Bằng việc kí hiệu θt là tham số đi kèm với t, xác suất P(t|c) được tính như sau:
+
+<br>
+
+
+**55. Remark: summing over the whole vocabulary in the denominator of the softmax part makes this model computationally expensive. CBOW is another word2vec model using the surrounding words to predict a given word.**
+
+&#10230; Chú ý: Cộng tổng tất cả các từ vựng trong mẫu số của phần softmax khiến mô hình này tốn nhiều chi phí tính toán. CBOW là một mô hình word2vec khác sử dụng các từ xung quanh để dự đoán một từ cho trước.
+
+<br>
+
+
+**56. Negative sampling ― It is a set of binary classifiers using logistic regressions that aim at assessing how a given context and a given target words are likely to appear simultaneously, with the models being trained on sets of k negative examples and 1 positive example. Given a context word c and a target word t, the prediction is expressed by:**
+
+&#10230; Negative sampling ― Nó là một tập của các bộ phân loại nhị phân sử dụng logistic regressions với mục tiêu là đánh giá khả năng mà một ngữ cảnh cho trước và các target words cho trước có thể xuất hiện đồng thời, với các mô hình đang được huấn luyện trên các tập của k negative examples và 1 positive example. Cho trước context word c và target word t, dự đoán được thể hiện bởi:
+
+<br>
+
+
+**57. Remark: this method is less computationally expensive than the skip-gram model.**
+
+&#10230; Chú ý: phương thức này tốn ít chi phí tính toán hơn mô hình skip-gram.
+
+<br>
+
+
+**57bis. GloVe ― The GloVe model, short for global vectors for word representation, is a word embedding technique that uses a co-occurence matrix X where each Xi,j denotes the number of times that a target i occurred with a context j. Its cost function J is as follows:**
+
+&#10230; GloVe - Mô hình GloVe, viết tắt của global vectors for word representation, nó là một kĩ thuật word embedding sử dụng ma trận đồng thời X với mỗi Xi,j là số lần mà target i xảy ra tại ngữ cảnh j. Cost function J của nó như sau:
```

Comment:
For all phrase "**co-occurence matrix**", translate it into "**Ma trận đồng xuất hiện**".

Reply:
GloVe - Mô hình GloVe, viết tắt của global vectors for word representation, nó là một kĩ thuật word embedding sử dụng ma trận đồng xuất hiện X với mỗi Xi,j là số lần mà từ đích (target) i xuất hiện tại từ ngữ cảnh (context) j. Cost function J của nó như sau: